OrcScala/src/orc/Backend.scala

```diff
@@ -80,13 +90,26 @@ trait Compiler[+CompiledCode] {
     override def toURL = toURI.toURL
   }
 
+  class ManyCompilationExceptions(exceptions: Seq[Throwable]) extends CompilationException("Multiple compiler errors") {
+    exceptions.foreach(addSuppressed)
+  }
+
   /** Compile the code in the reader using the given options and produce error messages on the
     * err writer. This is a simple wrapper around the other compile function.
     */
-  @throws(classOf[IOException])
+  @throws(classOf[IOException]) @throws(classOf[CompilationException])
```

Comment:
I'm concerned about this change.  We were trying to not throw exceptions for compile errors, although we never got completely away from them.  Have you checked that the Orchard services and the Eclipse plug-in still work with this change?